src/foam/u2/view/ScrollDAOView.js

```diff
@@ -341,9 +245,273 @@ foam.CLASS({
     }
   ],
 
+  classes: [
+    {
+      name: 'Row',
+      extends: 'foam.u2.Element',
+
+      documentation: `Recycled DOM rows. The "view" ViewSpec is used to manually
+          tear down and rebuild row contents whenever "data" changes. Scroll
+          performance is optimized by using an inner view with no dynamic
+          bindings so that the only DOM operation is the row completely
+          replacing its contents.`,
+
+      imports: [ 'rowFormatter' ],
+
+      axioms: [
+        foam.u2.CSS.create({
+          code: function CSS() {/*
+            ^ {
+              display: block;
+              contain: layout;
+              will-change: transform;
+              padding: 5px;
+              box-sizing: border-box;
+            }
+            ^ * {
+              background-color: #dddddd;
+            }
+        */}
+        })
+      ],
+
+      properties: [
+        [ 'nodeName', 'li' ],
+        {
+          name: 'data',
+          postSet: function(old, nu) {
+            if ( this.state !== this.LOADED ) return;
+            this.el().innerHTML = this.rowFormatter.format(nu);
```

Comment:
I still object to this being included in core framework code.  Some unsuspecting user is going to use it improperly, get a XSS vulnerability, and harm the reputation of the framework.

Please move ScrollDAOView to your own project.